.github/workflows/yocto-build-deploy.yml

```
@@ -0,0 +1,1174 @@
+      deploy-hostapp:
+        description: Whether to deploy a hostApp container image to a balena environment
+        required: false
+        type: boolean
+        default: true # we always want to deploy the hostapp - as eventuall we'll use it for HUP tests
+      finalize-hostapp:
+        description: Whether to finalize a hostApp container image to a balena environment
+        required: false
+        type: boolean
+        default: false # The default is "no" - because the `check-merge-tests` will determine whether or not to mark the hostapp as final. This is purely here for a manual override
+      check-merge-tests:
+        description: Whether to check the test results from the merge commit that resulted in new tagged version - can be overriden in dispatch for manual deploy
+        required: false
+        type: boolean
+        default: ${{ github.event_name == 'push' }} # This determines if we want to check the results of the merge PR - we only want to do it when a new tag is made
+      run-tests:
+        required: false
+        type: boolean
+        default: ${{ github.event_name == 'pull_request' }} # on pull request syncs + opens we want to run tests
+      deploy-ami:
+        description: Whether to deploy an AMI to AWS
+        required: false
+        type: boolean
+        default: false # For now always false, as it doesn't work. 
+      sign-image:
+        description: Whether to sign image for secure boot
+        required: false
+        type: boolean
+        default: false # Always false by default, override on specific device types which this is relevant in the device repo
+      os-dev:
+        description: Enable OS development features
+        required: false
+        type: boolean
+        default: false # Only for use with manual runs/dispatches
+      deploy-esr:
+        description: Enable to deploy ESR
+        required: false
+        type: boolean
+        default: ${{ startsWith(github.event.push.ref, 'refs/tags/v20') }} # Set this flag if the tag looks like an esr tag - this is good for 100 years or until balenaOS v20 comes out
+      aws_iam_role:
+        description: IAM role to assume for s3 permissions
+        required: false
+        type: string
+        default: "arn:aws:iam::567579488761:role/balena-os-deploy-s3-access-Role-8r0SXqfzgolr" #TODO - Replace using environments
+      aws_region:
+        description: AWS region
+        required: false
+        type: string
+        default: "us-east-1"
+      test-suites:
+        description: "comma-separated list of test suites to run. No spaces in between."
+        required: false
+        type: string
+        default: "os,hup,cloud"
+      test-workers:
+        description: "Comma-separated list of worker types to use for testing. No spaces in between. Example - qemu or testbot."
```

I think this is correct:

```suggestion
        description: "Comma-separated list of worker types to use for testing. No spaces in between. The valid worker types are `qemu` and `testbot`."
```